sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -360,9 +362,13 @@ public static ServiceBusReceivedMessage AmqpMessageToSBMessage(AmqpMessage amqpM
 
             if (amqpMessage.DeliveryTag.Count == GuidSizeInBytes)
             {
-                var guidBuffer = new byte[GuidSizeInBytes];
-                Buffer.BlockCopy(amqpMessage.DeliveryTag.Array, amqpMessage.DeliveryTag.Offset, guidBuffer, 0, GuidSizeInBytes);
-                sbMessage.LockTokenGuid = new Guid(guidBuffer);
+                Span<byte> guidBytes = stackalloc byte[GuidSizeInBytes];
+                amqpMessage.DeliveryTag.AsSpan().CopyTo(guidBytes);
```

Comment:
I may be overlooking something important about the raw form of the AMQP buffers, but I'm wondering why we need to make the intermediate copy.  

There is a `Guid` constructor that takes a `ReadOnlySpan<byte>` so we know it's not attempting to mutate the argument.  Would following would work?

```csharp
if (amqpMessage.DeliveryTag.Count == GuidSizeInBytes)
{
    sbMessage.LockTokenGuid= new Guid(amqpMessage.DeliveryTag.AsSpan());
}
```

If I'm not accounting for some oddity that `MemoryMarshal` handles, maybe a fallback case?

```csharp
if (amqpMessage.DeliveryTag.Count == GuidSizeInBytes)
{
    if (!MemoryMarshal.TryRead<Guid>(amqpMessage.DeliveryTag.AsSpan(), out var lockTokenGuid))
    {
        lockTokenGuid = new Guid(amqpMessage.DeliveryTag.Array);
    }

    sbMessage.LockTokenGuid = lockTokenGuid;
}
```

Reply:
The problem is netstandard. We can't properly multi target here to pick the appropriate guid overload. Or at least I tried and failed

Reply:
If we'd dual target netstandard 2 and 2.1 your proposal would work but that would be a whole other story I guess

Reply:
ah, thanks.  That's my bad.  I figured that I was overlooking something.  My test app was defaulted to .NET 5.